src/tla2528_adapters.cpp

```diff
@@ -0,0 +1,121 @@
+#include <libhal-util/bit.hpp>
+#include <libhal/error.hpp>
+#include <libhal/input_pin.hpp>
+#include <libhal/units.hpp>
+#include <libhal-expander/tla2528_adapters.hpp>
+
+namespace hal::expander {
+
+tla2528_output_pin make_output_pin(tla2528& p_tla2528,
+                                   hal::byte const p_channel,
+                                   hal::output_pin::settings const& p_settings)
+{
+  return tla2528_output_pin(p_tla2528, p_channel, p_settings);
+}
+tla2528_output_pin::tla2528_output_pin(
+  tla2528& p_tla2528,
+  hal::byte p_channel,
+  hal::output_pin::settings const& p_settings)
+  : m_tla2528(&p_tla2528)
+  , m_channel(p_channel)
+{
+  if (hal::bit_extract(hal::bit_mask::from(m_channel),
+                       m_tla2528->m_object_created)) {
+    throw hal::resource_unavailable_try_again(this);
+  }
```

Comment:
You use this same logic a few times in the adaptors. Would make sense to make a helper function for this.